### PR TITLE
[luci/import] Revise importModule method

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise importModule method with circle::Module as
- make it private to prevent future isses with large model
- add some guards
- only call reader with file data and size

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13678
- from #13615

**Review by jinevening — APPROVED** [review]
LGTM

File: compiler/luci/import/src/Importer.cpp [hunk 1]
```diff
@@ -268,6 +268,10 @@ Importer::Importer()
 
 std::unique_ptr<Module> Importer::importModule(const circle::Model *model) const
 {
+  assert(model);
```
Comment (jinevening) [inline compiler/luci/import/src/Importer.cpp:271]: This is checked in `importModule(const uint8_t *data, size_t size)`

Comment (seanshpark) [reply compiler/luci/import/src/Importer.cpp:271]: Thanks for pointing this :)

**Pull request merged**